src/main/java/nextstep/reservation/ReservationService.java

```diff
@@ -49,24 +51,74 @@ public Long create(Member member, ReservationRequest reservationRequest) {
+            throw new DataAccessException(DataAccessErrorCode.THEME_NOT_FOUND);
         }
 
         return reservationDao.findAllByThemeIdAndDate(themeId, date);
     }
 
     public void deleteById(Member member, Long id) {
-        Reservation reservation = reservationDao.findById(id);
-        if (reservation == null) {
-            throw new NullPointerException();
-        }
+        Reservation reservation = reservationDao.findById(id)
+                .orElseThrow(() -> new DataAccessException(DataAccessErrorCode.RESERVATION_NOT_FOUND));
 
         if (!reservation.sameMember(member)) {
-            throw new AuthenticationException();
+            throw new BusinessException(BusinessErrorCode.DELETE_FAILED_WHEN_NOT_MY_RESERVATION);
         }
 
         reservationDao.deleteById(id);
     }
+
+    public Long createWaiting(Member member, Long scheduleId) {
+        if (member == null) {
+            throw new AuthException(AuthErrorCode.INVALID_USER);
+        }
+        Schedule schedule = scheduleDao.findById(scheduleId)
+                .orElseThrow(() -> new DataAccessException(DataAccessErrorCode.SCHEDULE_NOT_FOUND));
+
+        Reservation newReservation = new Reservation(
+                schedule,
+                member
+        );
+
+        return reservationDao.save(newReservation);
+    }
+
+    public void deleteWaitingById(Member member, Long id) {
+        deleteById(member, id);
+    }
+
+    public List<Reservation> getReservationsByMember(Member member) {
+        if (member == null) {
+            throw new AuthException(AuthErrorCode.INVALID_USER);
+        }
+        List<Reservation> reservations = reservationDao.findByMemberId(member.getId());
+        return reservations.stream()
+                .filter(reservation -> {
+                    Long scheduleId = reservation.getSchedule().getId();
+                    Long waitTicketNumber = reservation.getWaitTicketNumber();
+                    return !isWaitingReservation(reservationDao.getPriority(scheduleId, waitTicketNumber));
+                })
+                .collect(Collectors.toList());
+    }
+
+    public List<ReservationWaitingResponseDto> getReservationWaitingsByMember(Member member) {
+        if (member == null) {
+            throw new AuthException(AuthErrorCode.INVALID_USER);
+        }
+        List<Reservation> reservationsByMemberId = reservationDao.findByMemberId(member.getId());
+        List<ReservationWaitingResponseDto> result = new ArrayList<>();
+        for (Reservation reservation : reservationsByMemberId) {
+            Long waitNum = reservationDao.getPriority(reservation.getSchedule().getId(), reservation.getWaitTicketNumber());
+            if (isWaitingReservation(waitNum)) {
+                result.add(ReservationWaitingResponseDto.of(reservation, waitNum));
+            }
+        }
+        return result;
+    }
+
+    private static boolean isWaitingReservation(Long waitNum) {
+        return waitNum != 0;
```

Comment:
Nit: 좀 더 명확해보여서 제안드립니다!
```suggestion
        return waitNum > 0;
```